src/services/pegout-data.processor.ts

```diff
@@ -1,27 +1,70 @@
 import {getLogger, Logger} from 'log4js';
+import {inject} from '@loopback/core';
 import {RskTransaction} from '../models/rsk/rsk-transaction.model';
 import {BRIDGE_METHODS, getBridgeSignature} from '../utils/bridge-utils';
 import FilteredBridgeTransactionProcessor from '../services/filtered-bridge-transaction-processor';
 import { BridgeDataFilterModel } from '../models/bridge-data-filter.model';
 import { PegoutStatusDataService } from './pegout-status-data-services/pegout-status-data.service';
+import { PegoutStatusDataModel } from '../models/rsk/pegout-status-data.model';
+import { PegoutStatusRulesService } from './pegout-status/pegout-status-rules-services';
+import {ServicesBindings} from '../dependency-injection-bindings';
 
 export class PegoutDataProcessor implements FilteredBridgeTransactionProcessor {
-  logger: Logger;
+  PegoutStatusRulesService: PegoutStatusRulesService;
   pegoutStatusDataService: PegoutStatusDataService;
+  logger: Logger;
 
-  constructor(pegoutStatusDataService: PegoutStatusDataService) {
+  constructor(
+    @inject(ServicesBindings.PEGOUT_STATUS_SERVICE)
+    PegoutStatusRulesService: PegoutStatusRulesService,
+    pegoutStatusDataService: PegoutStatusDataService)
+ {
     this.logger = getLogger('pegoutDataProcessor');
     this.pegoutStatusDataService = pegoutStatusDataService;
+    this.PegoutStatusRulesService = PegoutStatusRulesService;
   }
   async process(rskTransaction: RskTransaction): Promise<void> {
-    this.logger.error('[process] Method not yet implemented');
-    throw new Error('Method not yet implemented');
+    this.logger.debug(`[process] Got tx ${rskTransaction.hash}`);
+    const pegoutStatus = this.parse(rskTransaction);
+    if (!pegoutStatus) {
+      this.logger.debug('[process] Transaction is not a peg-out');
+      return;
+    }
+
+    try {
+      const found = await this.pegoutStatusDataService.getById(pegoutStatus.rskTxHash);
+      if (found) {
+        return this.logger.debug(`[process] ${rskTransaction.hash} already registered`);
+      }
+      await this.pegoutStatusDataService.set(pegoutStatus);
+      this.logger.trace(`[process] ${rskTransaction.hash} registered`);
+    } catch (e) {
+      this.logger.warn('[process] There was a problem with the storage', e);
+    }
   }
+
   getFilters(): BridgeDataFilterModel[] {
-    // TODO: add BRIDGE_METHODS.RELEASE_BTC = 'releaseBtc' when this method is available in the bridge abis.
     return [
       new BridgeDataFilterModel(getBridgeSignature(BRIDGE_METHODS.UPDATE_COLLECTIONS)),
       BridgeDataFilterModel.EMPTY_DATA_FILTER
     ];
   }
+
+  parse(transaction: RskTransaction): PegoutStatusDataModel | null {
```

Comment:
this seems like it would be specific parse, like `parsePegoutRequestTransaction`, although it lacks of the verification of the kind of transaction (via checking the filters)